lib/api/apiUtils/object/coldStorage.js

```diff
@@ -0,0 +1,107 @@
+const ObjectMDArchive = require('arsenal').models.ObjectMDArchive;
+const errors = require('arsenal').errors;
+const { config } = require('../../../Config');
+const { locationConstraints } = config;
+
+/**
+ * Check restore status.
+ *
+ * @param {ObjectMD} objectMD - object metadata
+ * @return {undefined}
+ */
+function _checkRestoreStatus(objectMD) {
+    if (objectMD.archive?.restoreRequestedAt && !objectMD.archive?.restoreCompletedAt) {
+        // check if archive.restoreRequestAt exists and archive.restoreCompletedAt not yet exists
+        return 'RestorePending';
+    } else if (objectMD.archive?.restoreCompletedAt
+        && new Date(objectMD.archive?.restoreWillExpireAt) >= new Date(Date.now())) {
+        //  check if archive.restoreWillExpireAt > current time
+        return 'Restored';
+    } else if (objectMD.archive?.restoreCompletedAt
+        && new Date(objectMD.archive?.restoreWillExpireAt) < new Date(Date.now())) {
+        return errors.InvalidObjectState;
+    } else if (objectMD.archive?.archiveInfo) {
+        return 'Archived';
+    } else {
+        return errors.InvalidObjectState;
+    }
+}
+
+/**
+ * Check if restore can be done.
+ *
+ * @param {BucketInfo} bucketMD - bucket instance
+ * @param {ObjectMD} objectMD - object metadata
+ * @return {Error|boolean} - true if the object is already restored
```

Comment:
Same issue here with the mixed return types. There doesn't seem to be a need for a boolean, as the different errors (of lack of error) already conveys the information.